autonomy/cli/build_images.py

```diff
@@ -44,28 +46,36 @@
     type=click.Path(dir_okay=True),
     help="Path to service dir.",
 )
+@click.option(
+    "-e",
+    "--extra-dependency",
+    "extra_dependencies",
+    type=PyPiDependency(),
+    help="Provide extra dependency.",
+    multiple=True,
+)
```

Comment:
You can provide extra dependencies while building the agent image using `-e` flag

```
$ autonomy build-image PUBLIC_ID -e open-aea-ledger-ethereum-flashbots==1.41.0
```